src/elmAnalyse.ts

```diff
@@ -0,0 +1,240 @@
+          const messages = state.messages;
+          messages.forEach(message => {
+            if (message.hasOwnProperty("data") && message.data.hasOwnProperty("type")) {
+              let messageType = message.data.type;
+              if (message.data.hasOwnProperty(messageType)) {
+                let messageTypeInfo = message.data[messageType];
+                let messageInfoFileSrc = messageTypeInfo.file;
+                let messageInfoFileRange = [0, 0, 0, 0];
+                if (messageTypeInfo.hasOwnProperty("range")) {
+                  messageInfoFileRange = messageTypeInfo.range.real;
+                }
+                let detail = "";
+                if (this.messageDescriptionsMap.has(messageType)) {
+                  detail = this.messageDescriptionsMap.get(messageType);
+                }
+                let issue: IElmIssue = {
+                  tag: 'analyser',
+                  overview: messageType,
+                  subregion: '',
+                  details: detail,
+                  region: this.correctRange(messageInfoFileRange),
+                  type: 'warning',
+                  file: path.join(cwd, messageInfoFileSrc)
+                };
+                this.elmAnalyseIssues.push(issue);
+              }
+            }
+          });
+          runLinter(vscode.window.activeTextEditor.document, this)
+        }
+        catch (e) {
+          vscode.window.showErrorMessage('Running websocket against Elm-analyse failed. Check if elm-analyse has been configured correctly.');
+          console.log("Socket not open error", e);
+        }
+      });
+      this.analyseSocket.on("error", (e) => {
+        vscode.window.showErrorMessage('Running websocket against Elm-analyse failed. Check if elm-analyse has been configured correctly.');
+        console.log("ERROR", e);
+      });
+    }
+    catch (e) {
+      vscode.window.showErrorMessage('Running websocket against Elm-analyse failed. If set to external - check if elm-analyse has been started in separate console.');
+      console.log("Could not start websocket against elm-analyse", e);
+    }
+  }
+
+  private startAnalyse(analyseCommand: string, analysePort: string, fileName: string, forceRestart = false) : void {
+
+    if (this.analyse.isRunning) {
+      vscode.window.showErrorMessage('Elm-analyse is already running. Please run the stop command if you want to restart elm-analyse.');
+    }
+    else {
+        this.analyse = execCmd(
+          analyseCommand,
+          {
+            fileName: fileName,
+            cmdArguments: ['-s', '-p ' + analysePort],
+            showMessageOnError: true,
+            onStart: () => this.analyse.stdin.write.bind(this.analyse.stdin),
+
+            onStdout: (data) => {
+              if (data) {
+                let info = data.toString();
+                this.oc.append(info);
+                if (info.match(/^Found \d* message/) && (this.analyseSocket === undefined || this.analyseSocket.CLOSED == 1)) {
```

Comment:
Not from the top of my head. To check if the correct server is running, we may do a HTTP request prior to connecting returning information of elm-analyse (version number, status, ect). If this call succeeds, then we may know if the server is correctly spawned. But such a mechanism is not in place at the moment.

For now you might better look for the output that matches this: `Listening on http://localhost:3000`.